tests/lax_test.py

```diff
@@ -3220,6 +3236,26 @@ def testBroadcastShapesReturnsPythonInts(self):
     out_shape = lax.broadcast_shapes(shape1, shape2)
     self.assertTrue(all(type(s) is int for s in out_shape))
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_shape={}_k={}_bdims={}".format(
+          jtu.format_shape_dtype_string(shape, dtype), k, bdims),
+       "shape": shape, "dtype": dtype, "k": k, "bdims": bdims, "rng_factory": rng_factory}
+      for shape in [(4,), (3, 4, 5)]
+      for k in [1, 3]
+      for bdims in all_bdims(shape)
+      # The top_k indices for integer arrays with identical entries won't match between
+      # vmap'd version and manual reference, so only test unique integer arrays for int_dtypes.
```

Comment:
Why is this?

Reply:
My guess was some stability issue, in the sense that if entries aren't unique then the manual reference might be stable where top_k isn't, or something.

Reply:
Both `TopK` and `TopKWithUnique` in the XLA client library use an `is_stable=true` sort (and shouldn't be otherwise unstable) on CPU/GPU, so I'd be surprised if there were unstable results here.

Reply:
Ah, good catch!

Reply:
So, a small repro of the underlying top_k instability issue is (public CPU colab):
```python
lax.top_k(jnp.array([[0, 0, 0, 1, 0], [5, 9, 8, 9, 4]]), k=2)[1]
# --> DeviceArray([[3, 0], [3, 1]], dtype=int32)
# same bottom row, small change on top row:
lax.top_k(jnp.array([[0, 0, 1, 0, 0], [5, 9, 8, 9, 4]]), k=2)[1]
# --> DeviceArray([[2, 0], [1, 3]], dtype=int32)
```
i.e. the values in other rows can effect the tie-breaking results.

Reply:
I think this is an XLA:CPU bug. I filed an internal XLA bug to track (b/155170120).

Reply:
Add a TODO?

Reply:
done